pip/utils/outdated.py

```diff
@@ -121,18 +123,21 @@ def pip_version_check(session):
 
         # Refresh the version if we need to or just see if we need to warn
         if pypi_version is None:
-            resp = session.get(
-                PyPI.pip_json_url,
-                headers={"Accept": "application/json"},
+            # Lets use PackageFinder to see what the latest pip version is
+            # This will allow local mirrors to be used for the check
+            finder = PackageFinder(
+                find_links=options.find_links,
+                index_urls=[options.index_url] + options.extra_index_urls,
+                # Old code use to effectively set this to False - Do we wish to maintain?
+                allow_all_prereleases=options.pre,
```

Comment:
Yea, we want to be able to publish pre-releases to PyPI without it triggering everyone to get told to upgrade their pip.